solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,522 @@
+    }
+
+    /**
+     * @dev returns the result of the conversion the provided tokens to the governance token and its burn
+     *
+     * @param tokens the tokens to convert
+     *
+     * @return the amounts of the converted tokens
+     * @return the network token amounts the tokens were converted to
+     * @return the total burned amount in this vortex
+     * @return the incentive fee resulting from this vortex
+     */
+    function availableVortex(IERC20[] calldata tokens)
+        external
+        view
+        returns (
+            uint256[] memory,
+            uint256[] memory,
+            uint256,
+            uint256
+        )
+    {
+        // check for duplicates in order to behave similarly to the vortex function
+        require(!hasDuplicates(tokens), "ERR_INVALID_TOKEN_LIST");
+
+        // retrieve conversion paths and the amounts to burn
+        (address[][] memory paths, uint256[] memory amounts, address[] memory govPath) =
+            vortexStrategy(tokens, networkFeeWallet());
+
+        // get all network token conversion amounts
+        BancorNetwork network = bancorNetwork();
+
+        uint256[] memory networkTokenConversionAmounts = new uint256[](tokens.length);
+        uint256 grossNetworkTokenConversionAmount = 0;
+        uint256 totalGovTokenAmountToBurn = 0;
+
+        for (uint256 i = 0; i < paths.length; ++i) {
+            // avoid empty conversions
+            uint256 amount = amounts[i];
+            if (amount == 0) {
+                continue;
+            }
+
+            address[] memory path = paths[i];
+            IERC20 token = IERC20(path[0]);
+            if (token == _govToken) {
+                // if the source token is the governance token, don't try to convert it either, but rather include it in
+                // the amount to burn
+                //
+                // Please note that networkTokenConversionAmounts[i] will remain 0 in this case
+                totalGovTokenAmountToBurn = totalGovTokenAmountToBurn.add(amount);
+            } else if (token == _networkToken) {
+                // if the source token is the network token, don't try to convert it, but rather add its amount to the
+                // total amount to convert to the governance token
+                grossNetworkTokenConversionAmount = grossNetworkTokenConversionAmount.add(amount);
+
+                networkTokenConversionAmounts[i] = amount;
+            } else {
+                // calculate the expected target amount
+                uint256 networkTokenConversionAmount = network.rateByPath(path, amount);
+                require(networkTokenConversionAmount > 0, "ERR_ZERO_TARGET_AMOUNT");
+
+                networkTokenConversionAmounts[i] = networkTokenConversionAmount;
+                grossNetworkTokenConversionAmount = grossNetworkTokenConversionAmount.add(networkTokenConversionAmount);
+            }
+        }
+
```

Comment:
I have a dejavu here.
Seems like a lot of duplicated code.
Perhaps we can move some pieces of code into common utility functions?

Reply:
We discussed this as well. It's hard to deduplicate this code because of the subtle difference embedded with (`convertByPath2` vs `rateByPath`, `ensureAllowance` and ETH handling, etc.). I always try to deduplicate code, but sometimes it's hard or adds unnecessary switching logic

Reply:
NP, remember that next time you ask me to unify the network-fee functions ;)

Reply:
It's a different story, but I will own your a favour in any case 😉